bindings/python/scripts/parse.py

```diff
@@ -1,145 +1,201 @@
+        - depth: The depth of the node (root=0)
 
+    Returns:
+        - parsed_info (dict): 
+            - Contains key-value pairs of various traits of a node
+            - The key 'members' contains the node's children's `parsed_info`
+    """
 
-def generate_parsed_info(cursor, filter, lines, this_filename, depth, parsed_list):
+    parsed_info = dict()
+
+    # Check if the cursor has spelling (cursor.spelling -> not NoneType)
     if cursor.spelling:
-        holder = {
-            "depth": depth,
-            "line": cursor.location.line,
-            "column": cursor.location.column,
-            "kind": cursor.kind.name,
-            "name": cursor.spelling,
-        }
+        parsed_info["depth"] = depth
+        parsed_info["line"] = cursor.location.line
+        parsed_info["column"] = cursor.location.column
+        parsed_info["kind"] = cursor.kind.name
+        parsed_info["name"] = cursor.spelling
         if cursor.type.kind.spelling != "Invalid":
-            holder["element_type"] = cursor.type.kind.spelling
+            parsed_info["element_type"] = cursor.type.kind.spelling
         if cursor.access_specifier.name != "INVALID":
-            holder["access_specifier"] = cursor.access_specifier.name
+            parsed_info["access_specifier"] = cursor.access_specifier.name
         if cursor.result_type.spelling != "":
-            holder["result_type"] = cursor.result_type.spelling
+            parsed_info["result_type"] = cursor.result_type.spelling
         if cursor.brief_comment:
-            holder["brief_comment"] = cursor.brief_comment
+            parsed_info["brief_comment"] = cursor.brief_comment
         if cursor.raw_comment:
-            holder["raw_comment"] = cursor.raw_comment
-        holder["members"] = []
-
-        parsed_list.append(holder)
+            parsed_info["raw_comment"] = cursor.raw_comment
+        parsed_info["members"] = []
 
+    # Get cursor's children and recursively add their info to the dictionary, as members of the parent
     for child in cursor.get_children():
-        if node_in_this_file(child, this_filename):
-            child_list = []
-            generate_parsed_info(
-                child, filter, lines, this_filename, depth + 1, child_list,
-            )
-            if child_list and parsed_list:
-                if len(parsed_list[0]["members"]):
-                    parsed_list[0]["members"].append(child_list[0])
-                else:
-                    parsed_list[0]["members"] = child_list
+        child_node = {"cursor": child, "filename": filename, "depth": depth + 1}
+        # Check if the child belongs to the file
+        if is_node_in_this_file(**child_node):
+            child_parsed_info = generate_parsed_info(**child_node)
+            # If both child and parent's info is not empty, add child's info to parent's members
+            if child_parsed_info and parsed_info:
```

Comment:
* Can the `parsed_info` be None and `child_parsed_info` be non None?
* Since this is `and`, we can move check for `parsed_info` outside the loop
* Since `parsed_info` depends only on `cursor.spelling`, we can have an early exit, resulting in out-denting of lines 83-98 (incoming)

Reply:
> Can the parsed_info be None and child_parsed_info be non None?

Yes for now. But that's an error since the parsed_info shouldn't be None #11. The `if cursor.spelling` shouldn't be checked. Needs #11 fixing.

> Can the parsed_info be None and child_parsed_info be non None?
> Since parsed_info depends only on cursor.spelling, we can have an early exit, resulting in out-denting of lines 83-98 (incoming)

Needs #11 fixing

Reply:
Added:
```
    """
    - compilation_commands:
        - An iterable object providing all the compilation commands available to build filename.
        - type: <class 'clang.cindex.CompileCommands'>
    - compilation_commands[0]:
        - Since we have only one command per filename in the compile_commands.json, extract 0th element
        - type: <class 'clang.cindex.CompileCommand'>
    - compilation_commands[0].arguments:
        - Get compiler arguments from the CompileCommand object
        - type: <class 'generator'>
    - list(compilation_commands[0].arguments)[1:-1]:
        - Convert the generator object to list, and extract compiler arguments
        - 0th element is the compiler name
        - nth element is the filename
    """
```

Reply:
> Can the parsed_info be None and child_parsed_info be non None?

Yes for now, but that's an error that needs resolving #11. Will update after #11. (The is `cursor.spelling` was used to ignore empty symbols, but the unnamed structs, enums, etc. requires special treatment)

> Since this is and, we can move check for parsed_info outside the loop
> Since parsed_info depends only on cursor.spelling, we can have an early exit, resulting in out-denting of lines 83-98 (incoming)

Updation after #11

Reply:
#15 
New PR.